src/mjolnir/ingest_transit.cc

```diff
@@ -426,10 +448,26 @@ bool write_stop_pair(Transit& tile,
     const bool origin_is_in_tile = origin_graphid_it != platform_node_ids.end();
     const bool dest_is_in_tile = dest_graphid_it != platform_node_ids.end();
 
+    // if it's not in the tile, we can't do anything else than take it's gtfs stop_id
+    // we check further down when stitching and adjust if it's was generated
+    std::string origin_onestop_id = origin_is_in_tile
+                                        ? tile_nodes.Get(origin_graphid_it->second.id()).onestop_id()
+                                        : get_origin_id_base(origin_stopId, feed_trip.feed);
+    std::string dest_onestop_id = dest_is_in_tile
+                                      ? tile_nodes.Get(dest_graphid_it->second.id()).onestop_id()
+                                      : get_origin_id_base(dest_stopId, feed_trip.feed);
+
```

Comment:
onestop_id equals e.g. `"Jena_stopid1"` if the origin/dest is not in this tile (note: this is actually wrong when it's generated, as it'd have its node_type appended). It gets its real onestop_id (including its node_type if generated) if it is in the tile.